prody/proteins/interactions.py

```diff
@@ -2358,9 +2359,48 @@ def showLigandInteraction_VMD(atoms, interactions, **kwargs):
     LOGGER.info("TCL file saved")
 
 
+def SMINA_extract_data(result):
+    """Supporting function for analysis of the SMINA results."""
+    import re
+    
+    data = {}
+
+    # Extracting Affinity from SMINA output
+    affinity_match = re.search(r'Affinity: ([0-9.-]+) \(kcal/mol\)', result)
+    if affinity_match:
+        data['Affinity'] = float(affinity_match.group(1))
+
+    # Extracting Intramolecular energy from SMINA output
+    intramolecular_energy_match = re.search(r'Intramolecular energy: ([0-9.-]+)', result)
+    if intramolecular_energy_match:
+        data['Intramolecular energy'] = float(intramolecular_energy_match.group(1))
+
+    # Extracting Weights and Terms from SMINA output
+    weights_terms_match = re.search(r'Weights\s+Terms\s*([\s\S]*?)## Name', result, re.DOTALL)
+    if weights_terms_match:
+        weights_terms_text = weights_terms_match.group(1)
+        term_lines = weights_terms_text.strip().split('\n')
+        term_dict = {}
+        for line in term_lines:
+            parts = line.split()
+            if len(parts) >= 2: 
+                weight = float(parts[0])
+                term = ' '.join(parts[1:])
+                term_dict[term] = weight
+        data.update(term_dict)
+        
+    term_values_match = re.search(r'Term values, before weighting:\n##\s+(.*?)\n', result, re.DOTALL)
+    if term_values_match:
+        term_values_text = term_values_match.group(1)
+        term_values_array = np.array([float(value) for value in term_values_text.split()])
+        data['Term values, before weighting'] = term_values_array.tolist()           
```

Comment:
You could maybe use the term_values_text for the keys here

Reply:
Why exactly? With the current code, we have access to all that we need.

Reply:
I don’t remember what I meant. I think I was looking for a way of accessing the keys and values together rather than separately. 

I’ll have another look tomorrow